src/API/Helpers/State/RedisStateHandler.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Auth0\SDK\API\Helpers\State;
+
+use Predis\Client;
+use Auth0\SDK\API\Helpers\State\StateHandler;
+
+/*
+ * This file is part of Auth0-PHP package.
+ *
+ * (c) Auth0
+ *
+ * For the full copyright and license information, please view the LICENSE file
+ * that was distributed with this source code.
+ */
+
+/**
+ * Redis based implementation of StateHandler.
+ *
+ * @author Auth0
+ */
+class RedisStateHandler implements StateHandler
+{
+    const STATE_NAME = 'webauth_state';
+
+    private $cache;
+
+    /**
+     * Auth0RedisStateHandler constructor.
+     *
+     * @param Client $cache
+     */
+    public function __construct(Client $cache)
+    {
+        $this->cache = $cache;
+    }
+
+    /**
+     * Generate state value to be used for the state param value during authorization.
+     *
+     * @return string
+     */
+    public function issue()
+    {
+        $state = uniqid('', true);
+        $this->store($state);
+        return $state;
+    }
+
+    /**
+     * Store a given state value to be used for the state param value during authorization.
+     *
+     * @param string $state
+     *
+     * @return mixed|void
+     */
+    public function store($state)
+    {
+        $this->cache->sadd(self::STATE_NAME, $state);
+    }
+
+    /**
+     * Perform validation of the returned state with the previously generated state.
+     *
+     * @param string $state
+     *
+     * @return boolean
+     */
+    public function validate($state)
+    {
+        $valid = $this->cache->sismember(self::STATE_NAME, $state) > 0;
```

Comment:
So this just checks a pool of values to make sure that the returned value exists in there? That's probably not going to fly, particularly because we're using `uniqid()`:

> This function does not guarantee uniqueness of return value. Since most systems adjust system clock by NTP or like, system time is changed constantly. Therefore, it is possible that this function does not return unique ID for the process/thread. 

https://www.php.net/manual/en/function.uniqid.php

Reply:
Yes, a UUID may be more appropriate here. I was just using most of the code from the session state handler, which uses uniqid.